src/components/Annotation.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  useContext,
+  useEffect,
+  useState,
+  useRef,
+} from 'react';
+import { createPortal } from 'react-dom';
+import MapContext from '../context/MapContext';
+import AnnotationProps from './AnnotationProps';
+
+export default function Annotation({
+  latitude,
+  longitude,
+
+  title = '',
+  subtitle = '',
+  accessibilityLabel = null,
+
+  selected = undefined,
+  onSelect = undefined,
+  onDeselect = undefined,
+  children,
+}: AnnotationProps) {
+  const [annotation, setAnnotation] = useState<mapkit.Annotation | null>(null);
+  const contentRef = useRef<HTMLDivElement>(document.createElement('div'));
```

Comment:
This is a little bit inefficient since it will call `document.createElement` on every render (even if the result will be ignored after the first render). Since we never mutate the object, we could use `useMemo` instead, which will do what we want. What do you think about this alternative?

Reply:
Really helpful! Ref didn't feel quite right, but I totally forgot about `useMemo`.